Source/Core/Core/HW/DVDInterface.cpp

```diff
@@ -329,32 +311,11 @@ static void FinishExecuteCommand(u64 userdata, int cyclesLate)
 	}
 }
 
-static void FinishExecuteReadCommand(u64 userdata, int cyclesLate)
-{
-	if (!current_read_command.is_valid)
-	{
-		PanicAlert("DVDInterface: There is no command to execute!");
-	}
-	else
-	{
-		// Here is the actual disc reading
-		if (!DVDRead(current_read_command.DVD_offset, current_read_command.output_address,
-		             current_read_command.length, current_read_command.decrypt))
-		{
-			PanicAlertT("Can't read from DVD_Plugin - DVD-Interface: Fatal Error");
-		}
-	}
-
-	// The command is marked as invalid because it shouldn't be used again
-	current_read_command.is_valid = false;
-
-	// The final step is to notify the emulated software that the command has been executed
-	CoreTiming::ScheduleEvent_Immediate(current_read_command.callback_event_type,
-	                                    current_read_command.interrupt_type);
-}
-
 static u32 ProcessDTKSamples(short *tempPCM, u32 num_samples)
 {
+	// TODO: Read audio data using the DVD thread instead of blocking on it?
```

Comment:
Yes, just push a streaming command to the dvd thread for it. Maybe also with the coretiming value as we'll need it for good audio dumping.